membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -122,178 +108,121 @@ case object Years extends UpToPeriodsType {
-    benefit: B,
-    billingPeriod: BP,
-    price: PricingSummary,
-    chargeId: ProductRatePlanChargeId,
-    subRatePlanChargeId: SubscriptionRatePlanChargeId,
-) extends RatePlanChargeList {
-  def benefits = NonEmptyList(benefit)
-}
+  def productType(catalog: Catalog): ProductType =
+    getCatalogZuoraPlan(catalog).productType
 
-/** Paper plans will have lots of rate plan charges, but the general structure of them is that they'll give you the paper on a bunch of days, and if
-  * you're on a plus plan you'll have a digipack
-  */
-case class PaperCharges(dayPrices: Map[PaperDay, PricingSummary], digipack: Option[PricingSummary]) extends RatePlanChargeList {
-  def benefits = NonEmptyList.fromSeq[Benefit](dayPrices.keys.head, dayPrices.keys.tail.toSeq ++ digipack.map(_ => Digipack))
-  def price: PricingSummary = (dayPrices.values.toSeq ++ digipack.toSeq).reduce(_ |+| _)
-  override def billingPeriod: BillingPeriod = BillingPeriod.Month
-  def chargedDays = dayPrices.filterNot(_._2.isFree).keySet // Non-subscribed-to papers are priced as Zero on multi-day subs
-  val subRatePlanChargeId = SubscriptionRatePlanChargeId("")
 }
 
-/** Supporter Plus V2 has two rate plan charges, one for the subscription element and one for the additional contribution.
+/** Low level model of a product rate plan, as it appears in the Zuora product catalog
   */
-case class SupporterPlusCharges(billingPeriod: BillingPeriod, pricingSummaries: List[PricingSummary]) extends RatePlanChargeList {
-
-  val subRatePlanChargeId = SubscriptionRatePlanChargeId("")
-  override def price: PricingSummary = pricingSummaries.reduce(_ |+| _)
-  override def benefits: NonEmptyList[Benefit] = NonEmptyList(SupporterPlus)
-}
-
-case class RatePlan(
-    id: RatePlanId,
-    productRatePlanId: ProductRatePlanId,
-    name: String,
-    description: String,
-    productName: String,
-    lastChangeType: Option[String],
-    productType: String,
-    product: Product,
-    features: List[SubsFeature],
-    charges: RatePlanChargeList,
-    chargedThrough: Option[
-      LocalDate,
-    ], // this is None if the sub hasn't been billed yet (on a free trial) or if you have been billed it is the date at which you'll next be billed
-    start: LocalDate,
-    end: LocalDate,
-)
-
-case class ProductRatePlan[+P <: Product, +C <: RatePlanChargeList, +S <: Status](
+case class ProductRatePlan(
     id: ProductRatePlanId,
-    product: P,
     name: String,
-    description: String,
-    saving: Option[Int],
-    charges: C,
-    s: S,
-)
+    productId: ProductId,
+    productRatePlanCharges: Map[ProductRatePlanChargeId, ProductRatePlanChargeProductType],
+    private val productTypeOption: Option[ProductType],
+) {
+  lazy val productType: ProductType = productTypeOption.getOrElse(throw new RuntimeException("Product type is undefined for plan: " + name))
+}
+
+// ProductType is the catalog product level ProductType__c
+case class ProductType(productTypeString: String) extends AnyVal
```

Comment:
this is as distinct from the one on the charge.  This wrapper just gives us a little more compile time type checking.